Add missing storage domain properties

maorlipchuk opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Add additional attributes on adding/import storage domain to oVirt.
The properties are :
    wipe_after_delete - Boolean flag which indicates whether the storage domain should wipe the data after delete.
    backup - Boolean flag which indicates whether the storage domain is configured as backup or not.
    critical_space_action_blocker - Inidcates what is the minimum free space the storage domain should contain.
    warning_low_space - Inidcates the minimum percentage of a free space in a storage domain to present a warning.

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request
 - New Module Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```

```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

Commit:
Add additional properties to storage domains

Commit:
add warn low space for additional storage properties

maorlipchuk commented:
@sivel 
@machacekondra

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py:
```diff
@@ -113,6 +113,21 @@
             - "C(lun_id) - LUN id."
             - "Note that these parameters are not idempotent."
+    discard_after_delete:
+        description:
```
please add `version_added: 2.5` to all new params

maorlipchuk replied:
done

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py:
```diff
@@ -113,6 +113,21 @@
             - "Note that these parameters are not idempotent."
+    discard_after_delete:
+        description:
+            - "Boolean flag which indicates whether the storage domain should discard after delete."
```
Can you please describe what `discard after delete` means? For what storages it's relevant etc.

maorlipchuk replied:
done

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py:
```diff
@@ -113,6 +113,21 @@
+            - "Boolean flag which indicates whether the storage domain is configured as backup or not."
+    critical_space_action_blocker:
+        description:
+            - "Inidcates what is the minimum free space the storage domain should contain."
```
In GB, in %?

maorlipchuk replied:
Added that it is in GB percentages

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py:
```diff
@@ -494,6 +524,11 @@ def main():
         posixfs=dict(default=None, type='dict'),
         glusterfs=dict(default=None, type='dict'),
         fcp=dict(default=None, type='dict'),
+        discard_after_delete=dict(type='bool', default=False),
```
Why not use default `None` and let backend default value?

maorlipchuk replied:
good point, will do

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py:
```diff
@@ -284,6 +307,13 @@ def build_entity(self):
+            discard_after_delete=self._module.params['discard_after_delete'],
+            backup=self._module.params['backup'],
+            critical_space_action_blocker=self._module.params['critical_space_action_blocker']
+            if self._module.params['critical_space_action_blocker'] else None,
```
Let's remove the `if` and set default `None`.

maorlipchuk replied:
done

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py:
```diff
@@ -284,6 +307,13 @@ def build_entity(self):
+            critical_space_action_blocker=self._module.params['critical_space_action_blocker']
+            if self._module.params['critical_space_action_blocker'] else None,
+            warning_low_space_indicator=self._module.params['warning_low_space']
+            if self._module.params['warning_low_space'] else None,
```
Let's remove the `if` and set default `None`.

maorlipchuk replied:
done

Commit:
Fixing comments

1. Fixing documentation
2. Use default None
3. Remove redundant if condition
4. remove added discard since it was already added

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py:
```diff
@@ -115,6 +115,22 @@
+        version_added: "2.5"
+    critical_space_action_blocker:
+        description:
+            - "Inidcates the minimal free space the storage domain should contain in GB percentages."
```
So it's in gigabytes or in percentage? it's not clear from the description.

maorlipchuk replied:
My bad, it is in percentage, will fix

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py:
```diff
@@ -545,6 +573,10 @@ def main():
         glusterfs=dict(default=None, type='dict'),
         fcp=dict(default=None, type='dict'),
+        wipe_after_delete=dict(type='bool', default=None),
+        backup=dict(type='bool', default=False),
```
Do we really want to have default `False` or do we want to have default `None` and let backend decide?

maorlipchuk replied:
sure, will change to None

Commit:
Apply comments #2

Fix default value to None
Use percentages instead of GB

machacekondra commented on lib/ansible/playbook/task.py:
```diff
@@ -414,17 +414,16 @@ def _get_parent_attribute(self, attr, extend=False, prepend=False):
         Generic logic to get the attribute or parent attribute for a task value.
```
Please remove this change.

machacekondra commented on test/integration/targets/iso_extract/tasks/main.yml:
```diff
@@ -35,7 +35,7 @@
   include_tasks: prepare.yml
```
Please remove this change.

maorlipchuk replied:
my bad, done

machacekondra reviewed (commented):
Please remove the non-relevant commit from core team.

machacekondra commented:
shipit

mwperina reviewed (approved):
shipit
